src/cls/tabular/cls_tabular.cc

```diff
@@ -2282,6 +2286,367 @@ int hep_query_op(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
+
+    op1.table_busy=false;
+    ::encode(op1, *out);
+    // TODO: encode new op and set
+    table_obj_map[op.table_name]=*out;
+    ret = cls_cxx_map_set_vals(hctx, &table_obj_map);
+    if (ret < 0)
+      return ret;
+    // encode result data for client.
+    //::encode(result_bl, *out);
+
+    return 0;
+}
+
+static
+int get_lock_obj_query_op(cls_method_context_t hctx, bufferlist *in, bufferlist *out)  
+{
+
+    inbl_lockobj_info op;
+
+    try {
+        bufferlist::iterator it = in->begin();
+        ::decode(op, it);
+    } catch (const buffer::error &err) {
+        CLS_ERR("ERROR: cls_tabular:init_lock_obj_query_op: decoding inbl_lockobj_op");
+        return -EINVAL;
+    }
+
+    std::string table_name = op.table_name;
+    int nobjs = op.num_objs;
+    bool tableBusy = op.table_busy;
+
+    CLS_LOG(20, "get_lock_obj_query_op: op.table_busy = %d", tableBusy);
+    CLS_LOG(20, "get_lock_obj_query_op: op.numobjs=%d", nobjs);
+    CLS_LOG(20, "get_lock_obj_query_op: op.table_name=%s", table_name);
+
+    using namespace Tables;
+    //schema_vec data_schema = schemaFromString(op.data_schema);
+
+
+    // Write to an omap instead
+
+    int ret;
+
+    bufferlist bl_entry;
+    ret = cls_cxx_map_get_val(hctx, table_name, &bl_entry);
+
+    if (ret < 0)
+      return ret;
+
+    inbl_lockobj_info op1;
+    try {
+        bufferlist::iterator it = bl_entry.begin();
+        ::decode(op1, it);
+    } catch (const buffer::error &err) {
+        CLS_ERR("ERROR: cls_tabular:init_lock_obj_query_op: decoding inbl_lockobj_op");
+        return -EINVAL;
+    }
+
+    CLS_LOG(20, "Get_lock_obj_query_op: op.table_busy = %d", op1.table_busy);
+    CLS_LOG(20, "Get_lock_obj_query_op: op.table_name = %s", op1.table_name);
+    CLS_LOG(20, "Get_lock_obj_query_op: op.table_group = %s", op1.table_group);
+    CLS_LOG(20, "Get_lock_obj_query_op: op.nobjs = %d", op1.num_objs);
+    // encode result data for client.
+    bufferlist result_bl;
+    result_bl.append("result data goes into result bl.");
+    //::encode(ret, *out);
+    ::encode(op1, *out);
+
+    
+
+    return 0;
+}
+
+static
+int acquire_lock_obj_query_op(cls_method_context_t hctx, bufferlist *in, bufferlist *out)  
+{
+
+    inbl_lockobj_info op;
+
+    try {
+        bufferlist::iterator it = in->begin();
+        ::decode(op, it);
+    } catch (const buffer::error &err) {
+        CLS_ERR("ERROR 1: cls_tabular:init_lock_obj_query_op: decoding inbl_lockobj_op");
+        return -EINVAL;
+    }
+
+    std::string table_name = op.table_name;
+    int nobjs = op.num_objs;
+    bool tableBusy = op.table_busy;
+
+    CLS_LOG(20, "acquire_lock_obj_query_op: op.table_busy = %d", tableBusy);
+    CLS_LOG(20, "acquire_lock_obj_query_op: op.numobjs=%d", nobjs);
+
+    using namespace Tables;
+    //schema_vec data_schema = schemaFromString(op.data_schema);
+
+
+    // Write to an omap instead
+
+    int ret;
+
+    bufferlist bl_entry;
+    ret = cls_cxx_map_get_val(hctx, table_name, &bl_entry);
```

Comment:
add note this is getting the current status (lock oop struct) of the lock for this table.